include/mbedtls/check_config.h

```diff
@@ -298,6 +298,10 @@
 #error "MBEDTLS_MEMORY_BUFFER_ALLOC_C defined, but not all prerequisites"
 #endif
 
+#if defined(MBEDTLS_MEMORY_BACKTRACE) && !defined(MBEDTLS_MEMORY_BUFFER_ALLOC_C)
+#error "MBEDTLS_MEMORY_BACKTRACE defined, but not all prerequesites"
+#endif
+
```

Comment:
`MBEDTLS_MEMORY_DEBUG` also requires `MBEDTLS_MEMORY_BUFFER_ALLOC_C` in the sense that it has no effect without it.

Reply:
Addressed in f0fde48.